allowed.py

```diff
@@ -367,10 +368,19 @@ def get_constructs(last_unit: int) -> tuple:
     )
 
 
-def check_tree(tree: ast.AST, constructs: tuple, source: list) -> list:
+def check_tree(
+    tree: ast.AST,
+    constructs: tuple,
+    source: list,
+    from_ipynb: bool,
+    line_cell_map: dict,
```

Comment:
The boolean seems unnecessary because if it's false the dictionary is empty. I guess that a notebook without code cells also leads to an empty dict, but in that case there's nothing to check and report.

Reply:
Yes, good point, it does seem unnecessary. At the time it did cross my mind to use the empty dict test, but I had convinced myself there was some case where it would cause a problem, probably the case you have mentioned, but as you say, if the dict is empty (and the file is a notebook) then there will be no checks and no errors!